### ISSUE TITLE
Vue.js throws warning when I run Jest test

### ISSUE THREAD

**Issue by sumerokr** [mswjs/msw#506]
## Describe the bug
Out-of-the-box Vue CLI based project. Added Jest and mswjs.
I tried 3 versions of node-based fetchers (axios, node-fetch, whatwf-fetch).
whatwf-fetch works fine, but tests based on axios and node-fetch throws a warning/error in the console.
```
[Vue warn]: Error in nextTick: "TypeError: Cannot read property 'createElement' of null"
```

Browser works without any issue. Only server implementation has this problem.

## Environment

- `msw: 0.24.2`
- `nodejs: 12.19.0`
- `npm: 6.14.8`
- `Chrome: 87.0.4280.88 (Official Build) (64-bit)`

## To Reproduce
1. Clone https://github.com/sumerokr/vue-mswjs
2. Checkout branch `axios` or `node-fetch`
3. Run `npm run test:unit`

## Expected behavior
No warning/error in the console after the tests

P. S.
Try to uncomment one more test in spec file. Both works!
Try to remove `v-if` condition in Vue template. Works!
Try to avoid mswjs based endpoint and use the real one. Works!

I can't figure out, what is wrong here. But this error makes it uncomfortable to use mswjs in Vue projects.

**Comment by timdeschryver** [comment]
Here's what I could find. 

I think this is because the request resolves after/while the test is finished (this can be tested by logging the response in the component). When you use a real web request, you don't see any logs because I assume the request gets canceled (the previous logged response won't be logged anymore). 

Because of this, an option would be that MSW shouldn't emit any "pending" requests when  the server is closed.
But this leaves us with the question, what do we do with `resetHandlers`?

The error should also goes away if you await the response in the test itself.

**Comment by AeonFr** [comment]
I have customized the `afterEach` hook in jest to give warnings when a test keeps a pending request open. It also uses `waitFor` to give some margin for the request to close itself, and, just in case, uses `Vue.$nextTick` to make sure no template renders are scheduled before the test finishes.

Note this solution will only work if you run jest with the `--runInBand` flag, otherwise the openRequest count will not be accurate (I think). Also note you need to be on the latest version of `msw` for this to work, since `server.on()` is a relatively new feature.

```
// ensure no requests are left pending between tests
import { waitFor } from "@testing-library/vue";
import Vue from "vue";

let openRequests = 0;
server.on("request:start", (req) => {
  openRequests++;
});

server.on("request:end", (req) => {
  openRequests--;
});

afterEach(async () => {
  // wait for all requests to end
  await waitFor(() => {
    return openRequests === 0;
  }, 5000);

  if (openRequests > 0) {
    console.warn(
      "\x1b[0m\x1b[1m%s\x1b[0m\x1b[34m%s\x1b[0m\x1b[2m%s\x1b[0m", // colors!
      `Your test "${
        expect.getState().currentTestName
      }" has finished with pending requests. Since store instance's are shared between tests, this might have side-effects.`,
      "\nEnsure you're mocking all requests and waiting for them to finish. Add async assertions (findBy*, waitFor) if necessary.",
      "\n(Note: this warning will always get reported when a test fails and exits in the middle of a series of API calls)"
    );
  }

  // wait a "tick" to allow for gracefull cleanup of Vue async callbacks
  await Vue.nextTick();

  // Reset any request handlers that we may add during the tests,
  // so they don't affect other tests.
  server.resetHandlers();
});
```

The warning looks something like this:

![Screenshot 2021-01-07 at 15 36 31](https://user-images.githubusercontent.com/11742754/103904850-1e961680-50fe-11eb-9e8a-60407903ade6.png)

**Comment by kettanaito** [comment]
Can you please try to reproduce this issue with the latest library release (0.27.0)?

**Comment by sumerokr** [comment]
@kettanaito I have no good news, unfortunately. The problem is still here, even with the 0.27.0 version.
I used the repo and branches mentioned in the first message to reproduce the issue.

**Comment by pirhoo** [comment]
Hey guys, I'm getting the same bug when using MSW with axios.

Everything works as excepted but I'm still getting this weird Vue warning.

Any chance you found a workaround?

```
console.error node_modules/vue/dist/vue.runtime.common.dev.js:621
  [Vue warn]: Error in nextTick: "TypeError: Cannot read property 'createElement' of null"

console.error node_modules/vue/dist/vue.runtime.common.dev.js:1893
  TypeError: Cannot read property 'createElement' of null
      at Object.createElement$1 [as createElement] (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:5709:22)
      at createElm (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:5973:19)
      at VueComponent.patch [as __patch__] (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:6499:7)
      at VueComponent.Vue._update (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:3948:19)
      at VueComponent.updateComponent (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:4069:10)
      at Watcher.get (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:4481:25)
      at new Watcher (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:4470:12)
      at mountComponent (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:4076:3)
      at VueComponent.Object.<anonymous>.Vue.$mount (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:8436:10)
      at init (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:3131:13)
      at createComponent (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:6002:9)
      at createElm (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:5949:9)
      at updateChildren (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:6240:11)
      at patchVnode (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:6343:29)
      at updateChildren (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:6217:9)
      at patchVnode (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:6343:29)
      at VueComponent.patch [as __patch__] (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:6504:9)
      at VueComponent.Vue._update (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:3951:19)
      at VueComponent.updateComponent (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:4069:10)
      at Watcher.get (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:4481:25)
      at Watcher.run (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:4556:22)
      at flushSchedulerQueue (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:4314:13)
      at Array.<anonymous> (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:1985:12)
      at flushCallbacks (<REDACTED>/node_modules/vue/dist/vue.runtime.common.dev.js:1911:14)
```

## Environement

* `nodejs: 14.5.15`
* `msw: 0.33.2`
* `jest: 24.9.0`
* `vue: 2.6.11`

**Comment by kettanaito** [comment]
The issue seems to be solved as of `0.40.2`. I have no exact version at which this as been resolved. 

Here's the output of of `npm run test:unit` in the mentioned reproduction repo:

```sh
vue-mswjs % npm run test:unit

> vue-mswjs@0.1.0 test:unit
> vue-cli-service test:unit

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
 PASS  tests/unit/example.spec.js
  HelloWorld.vue
    ✓ renders props.msg when passed (9ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.31s
Ran all test suites.
```

There are no warnings or errors, the test passes without issues.
